src/lib/functions/server.js

```diff
@@ -38,13 +38,22 @@ const buildClientContext = function (headers) {
   }
 }
 
+// same as https://github.com/netlify/bitballoon/blob/fbd7881e6c8e8c48e7a0145da4ee26090c794108/app/models/deploy.rb#L482
+const isValidNetlifyFunctionName = (name) => /^[\w-]+$/.test(name)
+
 const createHandler = function ({ functionsRegistry }) {
   return async function handler(request, response) {
     // handle proxies without path re-writes (http-servr)
     const cleanPath = request.path.replace(/^\/.netlify\/functions/, '')
     const functionName = cleanPath.split('/').find(Boolean)
     const func = functionsRegistry.get(functionName)
 
+    if (!isValidNetlifyFunctionName(functionName)) {
```

Comment:
There's two checks we make here:

1. Does this function exist? (is there a corresponding file?)
2. Is the function name valid? (regex test)

In which order should we perform them? Should we error for invalid names, even if the function doesn't exist? Or should we only check the name if there's a corresponding file? Is one preferable DX-wise?

Reply:
I think we should check for the existence of the file first, and only then check whether it's a valid function name. If you expect a function to exist and it doesn't, I feel that's the problem you need to address first, regardless of whether you're choosing a valid name for it.

Reply:
addressed in 80749db